### DISCUSSION TITLE
.bok file extension.

### DISCUSSION THREAD

**Discussion by Zeyadas** [koreader/koreader#7971] [Q&A]
I don't remember if I asked this before, but can koreader open .bok file extension? Or can you add this file extension?
[An example, direct download .rar file ](https://shamela.ws/books/205/20587.rar)

**Comment by Frenzie** [comment]
I wouldn't count on it, but we'd need a specification of the format to give a more definitive answer. Here are a few quick links I found.

https://sourceforge.net/projects/miftah/ 
https://shamela.org/
https://shamela.ws/

**Reply by Zeyadas** [reply]
I'm not a programmer, and I don't know a lot about file extensions but I think the .bok file extension is similar to the epub..

**Reply by poire-z** [reply]
It does not look like it is similar to EPUB.
EPUB are ZIP files.

Your .bok is detected as a Microsoft Access Database.
The miftah.tar source code from one of the above links contains a python file that starts with:
```py
# AccessDump.py
# A simple script to dump the contents of a Microsoft Access Database.
# It depends upon the mdbtools suite:
#   http://sourceforge.net/projects/mdbtools/
# Website:
#   http://www.guyrutenberg.com/2012/07/16/sql-dump-for-ms-access-databases-mdb-files-on-linux/
# MDB_JET3_CHARSET="cp1256" python AccessDump.py filename.bok | sqlite3 filename.mfth
```
to convert it to some sqlite database. (And what you get in the sqlite database is not clear at all :)
It's very very different from all the kinds of documents we currently support, and some very old database format we really don't want to have to support.

So, you better try to find some way to convert them to some other more popular format, like PDF.

**Reply by Zeyadas** [reply]
Now I understand.. Thank you.
